packages/firebase_core/firebase_core_web/lib/firebase_core_web.dart

```diff
@@ -59,11 +58,11 @@ String _getJSErrorCode(dynamic e) {
 /// If a JavaScript error is thrown and not manually handled using the code,
 /// this function ensures that if the error is Firebase related, it is instead
 /// re-created as a [FirebaseException] with a familiar code and message.
-FirebaseException _catchJSError(dynamic e) {
-  if (js_util.getProperty(e, 'name') == 'FirebaseError') {
-    String rawCode = js_util.getProperty(e, 'code');
+FirebaseException _catchJSError(JSObject e) {
+  if (e.getProperty<JSString?>('name'.toJS)?.toDart == 'FirebaseError') {
```

Comment:
This is valid code, but I'd use an interface to make this more readable. Here's an example:

```dart
@JS()
@staticInterop
class JSError {}

extension on JSError {
  external String? get name;
  external String? get message;
  external String? get code;
} 
```

which would let you write simpler code like:

`if (e.name == 'FirebaseError')`

`return r.code ?? ''`

and so forth. You may not have access to extension types since that will require `>=3.3`, but it's even more concise with one:

```dart
extension type Error(JSObject e) {
  external String? get name;
  external String? get message;
  external String? get code;
}
```

Reply:
Thanks I've updated this as well.
We don't have access to Dart 3.3 yet, but I'll keep in mind to revisit this once Flutter supports it